fvm/evm/handler/handler.go

```diff
@@ -267,6 +276,13 @@ func (h *ContractHandler) commitBlockProposal() error {
 		return err
 	}
 
+	// report metrics
+	h.backend.EVMBlockExecuted(
+		len(bp.TxHashes),
+		bp.TotalGasUsed,
+		types.UnsafeCastOfBalanceToFloat64(bp.TotalSupply),
```

Comment:
should the cast be done by the metrics collector?

Reply:
I first put it there but then it had to have dependency to Cadence internal logic